src/allmydata/test/test_istorageserver.py

```diff
@@ -1046,18 +1086,80 @@ def disconnect(self):
         assert self.storage_client is not current
 
 
+class _HTTPMixin(_SharedMixin):
+    """Run tests on the HTTP version of ``IStorageServer``."""
+
+    def setUp(self):
+        if PY2:
+            self.skipTest("Not going to bother supporting Python 2")
+        return _SharedMixin.setUp(self)
+
+    def _get_istorage_server(self):
+        set_treq_pool(HTTPConnectionPool(reactor, persistent=False))
+        swissnum = b"1234"
+        self._http_storage_server = HTTPServer(self.server, swissnum)
+        self._port_number = allocate_tcp_port()
+        self._listening_port = reactor.listenTCP(
```

Comment:
It might be cleaner and is almost certainly more correct to use `self.addCleanup` rather than `tearDown` to shut down this listening port.  If the test somehow fails after setUp finishes but before this port is bound then tearDown will run and fail because there is no `_listening_port` attribute, I think.

Also, maybe this function should guard against being called twice since in that case I think things will start to get pretty confusing if it goes ahead and sets up a second instance of everything, clobbering the record of the first.

Reply:
I switched to `addCleanup`, and stopped setting anything on `self`, which should prevent anything from being clobbered if mistakenly called twice.